rust/actyx/ax/src/cmd/apps/license.rs

```diff
@@ -91,113 +110,31 @@ fn parse_expires_in_as_duration(expires_in: &str) -> Result<chrono::Duration, an
         .captures(expires_in)
         .ok_or_else(|| anyhow::anyhow!("Failed to parse string."))?;
 
+    let get_i64_from_capture = |name: &str| {
+        captures
+            .name(name)
+            .and_then(|m| m.as_str().parse::<i64>().ok())
+            .unwrap_or(0)
+    };
+
     let mut duration = chrono::Duration::zero();
-    duration = duration + chrono::Duration::days(365 * get_i64_from_capture(&captures, "years"));
-    duration = duration + chrono::Duration::days(30 * get_i64_from_capture(&captures, "months"));
-    duration = duration + chrono::Duration::days(7 * get_i64_from_capture(&captures, "weeks"));
-    duration = duration + chrono::Duration::days(get_i64_from_capture(&captures, "days"));
-    duration = duration + chrono::Duration::hours(get_i64_from_capture(&captures, "hours"));
-    duration = duration + chrono::Duration::minutes(get_i64_from_capture(&captures, "minutes"));
-    duration = duration + chrono::Duration::seconds(get_i64_from_capture(&captures, "seconds"));
+    duration = duration + chrono::Duration::days(365 * get_i64_from_capture("years"));
+    duration = duration + chrono::Duration::days(30 * get_i64_from_capture("months"));
+    duration = duration + chrono::Duration::days(7 * get_i64_from_capture("weeks"));
+    duration = duration + chrono::Duration::days(get_i64_from_capture("days"));
+    duration = duration + chrono::Duration::hours(get_i64_from_capture("hours"));
+    duration = duration + chrono::Duration::minutes(get_i64_from_capture("minutes"));
+    duration = duration + chrono::Duration::seconds(get_i64_from_capture("seconds"));
```

Comment:
I'm not a fan of this approach because it obscures where the captures are coming from. 
It requires one extra reading step to see that the `captures` are captured by the closure, instead of being very obvious where they come from.

Reply:
My view is inspired by the same goal (readability) but diametrically different: I want to see as few redundant glyphs as possible so that the human eye concentrates on the essence of the differences between the lines. I’d prefer something like:

    duration += chrono::days(365 * get("years"));

Reply:
Agreed on the `+=` but that's a Chrono issue (see https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3c541ba510954293916695f4397848e7)
I don't have a strong opinion on `get` vs  `get_i64_from_capture`. In these small cases it's always hard to vouch for one over the other without more people providing an opinion for better sampling :sweat_smile:

Reply:
Since `chrono::Duration` is indeed lacking I made an even more tailored utility function. I think those are fine if they easily fit on the same screen of their strictly scoped usage — which is also why I keep them inside the outer function instead of lifting them to the module level.